### DISCUSSION TITLE
Einbindung Felicity Speicher BMS

### DISCUSSION THREAD

**Discussion by pade1984** [hoylabs/OpenDTU-OnBattery#1119] [Ideas]
Hallo,

ich nutze seit ca. einer Woche openDTU on Battery mit zwei Victron MPPTs, Victron Smartshunt und einem 5,1 kWh Speicher von Felicity Solar. Da der SoC zwischen dem Felicity BMS und dem Smartshunt selbst innerhalb eines Tages teilweise um bis zu 5 % auseinander liegen, wäre es super, wenn das BMS direkt eingelesen werden könnte. Felicity unterstützt CAN und RS485. Leider habe ich keine Ahnung wie man das einbinden könnte. Könnt ihr da ggf. helfen?

**Comment by SW-Niko** [comment]
Hallo @pade1984,
das wird schwierig. Da müssen wir einen Programmierer finden der:

1.  So einen Felicity Speicher hat
2. Der Hersteller die Befehlsliste der Schnittstelle offen legt
3. Programmieren kann
4. Zeit und Lust hat 

Aber ... es finden sich immer wieder Wege und Leute die so was angehen.
Also nicht sauer sein, wenn du wenig Reaktion auf deine Anfrage erhältst. 😬

**Reply by Bootino** [reply]
>  [Felicity Registers.ods](https://1drv.ms/x/c/91afadfff262524e/ETa3im_aSsVEqHh1BZC5nw4B72agq-jey1KqXxEpfrVB8Q?e=1qRjlf)
> 
> Das ist die von Felicity zur Verfügung gestellte Modbus Datei, übersetzt von Felicity aus dem Chinesischen ins Englische und damit lesbar.
> 
> Wie gesagt, der Modbus ist via COM-Port via Pins 3 & 4 verfügbar.
> 


Quelle: https://community.simon42.com/t/felicity-solar-per-api-abfragen/42754/11
hilft das weiter?

**Reply by moinmoin-sh** [reply]
Felicity funktioniert mit openDTUoB "out of the box" über CAN verbunden und als "Pylontech" konfiguriert.

![grafik](https://github.com/user-attachments/assets/69e77821-f99a-45a9-a7f6-367cb761108a)

![grafik](https://github.com/user-attachments/assets/828ce72e-5942-42c5-b985-506bb8d6af6c)

**Reply by Jaggerman-Jensen** [reply]
Hallo zusammen,
habe mir auch eine Felicity 12,5kWh Batterie zugelegt. Hast du das Kabel so konfektioniert, dass Orange/weiß (CANL) auf Pin 7 und Orange (CANH) auf Pin 8 vom RJ45 Stecker geht? Werde mich am Wochenende an die Installation begeben. 
Eine Frage habe ich zudem: Welche Dip-Schalter Einstellung ist bei einer Batterie die richtige? Im Manual gibt es 2 Tabellen dazu. Einmal wäre es Dip 1,5 , bei der 2. Tabelle alle auf 0 und im Auslieferungszustand war der 1. on... 😵‍💫
Würde mich sehr über Unterstützung freuen!
Beste Grüße 
Jens

![IMG_20250603_153617](https://github.com/user-attachments/assets/b7120a44-870d-4f81-946f-003235b60dfb)
![IMG_20250603_153557](https://github.com/user-attachments/assets/33f02d6f-8fe9-4da8-8117-29947059739f)

**Reply by moinmoin-sh** [reply]
Hallo @Jaggerman-Jensen 

> Hast du das Kabel so konfektioniert, dass Orange/weiß (CANL) auf Pin 7 und Orange (CANH) auf Pin 8 vom RJ45 Stecker geht?
> 

Ja, an meiner Batterie gibt es es noch die RJ45 Buchse `PCS ` die ich für die CAN Verbindung zu openDTUoB benutze

>  Welche Dip-Schalter Einstellung ist bei einer Batterie die richtige? Im Manual gibt es 2 Tabellen dazu. Einmal wäre es Dip 1,5 , bei der 2. Tabelle alle auf 0 und im Auslieferungszustand war der 1. on... 
> 

Bei mir ist auch Dip1 auf on.
lt. deiner Doku Seite 15 (s.o.) Bemerkung 3 ist für eine einzelne Batterie aber dip 1-5 auf off auch richtig. 
Dip 5 ist nur für die CAN Terminierung der CAN Kommunikation der Batterien untereinander erforderlich wenn mehrere Batterien parallel geschaltet sind. (muss dann bei der ersten und letzten Batterie eingeschaltet sein)

Viel Erfolg!

**Comment by AndreasBoehm** [comment]
Hallo @pade1984 

Wenn du deinen Speicher via CAN verbindest (Pylontech oder Pytes auswählen) und "Ausführliche Protokollierung" aktivierst kannst du die eingehenden CAN Messages des Speichers in der Konsole ansehen und hier posten.

Vielleicht haben wir Glück und es findet sich ein ähnliches CAN Protokoll im Internet.

Leider geht das aktuell noch nicht in einem Release build, ich kann dir aber gerne ein build verlinken wenn du nicht selber bauen willst.

**Comment by schlimmchen** [comment]
> Victron Smartshunt

Vielleicht reicht es dir auch, wenn du den DPL auf Basis von Spannungen einstellst (so tue ich es auch, weil auch mit der SoC zu ungenau ist und er triftet). Die genaue Spannung des Akkus erhälst du ja vom SmartShunt.
